lib/engines/content_block_manager/app/controllers/concerns/workflow/update_methods.rb

```diff
@@ -0,0 +1,76 @@
+module Workflow::UpdateMethods
+  extend ActiveSupport::Concern
+
+  REVIEW_ERROR = Data.define(:attribute, :full_message)
+
+  UPDATE_ACTIONS = {
+    review_links: :redirect_to_schedule,
+    schedule_publishing: :validate_schedule,
+    internal_note: :update_internal_note,
+    change_note: :update_change_note,
+    review_update: :validate_review_page,
+    review: :validate_review_page,
+  }.freeze
+
+  def redirect_to_schedule
+    redirect_to content_block_manager.content_block_manager_content_block_workflow_path(
+      id: @content_block_edition.id,
+      step: :schedule_publishing,
+    )
+  end
+
+  def validate_schedule
+    @content_block_edition = ContentBlockManager::ContentBlock::Edition.find(params[:id])
+
+    validate_scheduled_edition
+
+    redirect_to content_block_manager.content_block_manager_content_block_workflow_path(
+      id: @content_block_edition.id,
+      step: :internal_note,
+    )
+  rescue ActiveRecord::RecordInvalid
+    render "content_block_manager/content_block/editions/workflow/schedule_publishing"
+  end
+
+  def update_internal_note
+    @content_block_edition.update!(internal_change_note: edition_params[:internal_change_note])
+
+    redirect_to content_block_manager.content_block_manager_content_block_workflow_path(
+      id: @content_block_edition.id,
+      step: :change_note,
+    )
+  end
+
+  def update_change_note
+    @content_block_edition.assign_attributes(change_note: edition_params[:change_note], major_change: edition_params[:major_change])
+    @content_block_edition.save!(context: :change_note)
+
+    redirect_to content_block_manager.content_block_manager_content_block_workflow_path(
```

Comment:
the back link isn't working when i hit an error on the change note form, does that happen for you?

Reply:
Ah, it might do actually. I'll fix that